segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/ZooKeeperServiceRunner.java

```diff
@@ -106,19 +163,134 @@ public void stop() {
+            System.clearProperty("zookeeper.ssl.keyStore.location");
+            System.clearProperty("zookeeper.ssl.keyStore.password");
+            System.clearProperty("zookeeper.ssl.trustStore.location");
+            System.clearProperty("zookeeper.ssl.trustStore.password");
+        }
     }
 
     /**
      * Blocks the current thread and awaits ZooKeeper to start running locally on the given port.
      *
      * @param zkPort The ZooKeeper Port.
+     * @param secureZk Flag to notify whether the ZK is secure.
+     * @param trustStore Location of the trust store.
+     * @param keyStore Location of the key store.
+     * @param keyStorePasswordPath Location of password path for key store.
+     *                             Empty string if `secureZk` is false or a password does not exist.
+     * @param trustStorePasswordPath Location of password path for trust store.
+     *                               Empty string if `secureZk` is false or a password does not exist.
      * @return True if ZooKeeper started within a specified timeout, false otherwise.
      */
+    public static boolean waitForServerUp(int zkPort, boolean secureZk, String trustStore, String keyStore,
+                                          String keyStorePasswordPath, String trustStorePasswordPath) {
+        val address = "localhost:" + zkPort;
+        if (secureZk) {
+            return waitForSSLServerUp(address, LocalBookKeeper.CONNECTION_TIMEOUT, trustStore, keyStore,
+                    keyStorePasswordPath, trustStorePasswordPath);
+        } else {
+            return LocalBookKeeper.waitForServerUp(address, LocalBookKeeper.CONNECTION_TIMEOUT);
+        }
+    }
+
     public static boolean waitForServerUp(int zkPort) {
-        val address = LOOPBACK_ADDRESS.getHostAddress() + ":" + zkPort;
-        return LocalBookKeeper.waitForServerUp(address, LocalBookKeeper.CONNECTION_TIMEOUT);
+        return waitForServerUp(zkPort, false, "", "", "", "");
+    }
+
+    private static boolean waitForSSLServerUp(String address, long timeout, String trustStore, String keyStore,
+                                              String keyStorePasswdPath, String trustStorePasswordPath) {
+        long start = MathUtils.now();
+        String[] split = address.split(":");
+        String host = split[0];
+        int port = Integer.parseInt(split[1]);
+
+        while (true) {
+            try {
+                SSLContext context = SSLContext.getInstance("TLS");
+                TrustManagerFactory trustManager = getTrustManager(trustStore, trustStorePasswordPath);
+                KeyManagerFactory keyFactory = getKeyManager(keyStore, keyStorePasswdPath);
+
+                context.init(keyFactory.getKeyManagers(), trustManager.getTrustManagers(), null);
+                SSLSocketFactory factory = context.getSocketFactory();
+
+                Socket sock = context.getSocketFactory().createSocket(new Socket(host, port), host, port, true);
```

Comment:
Can you use `try-with-resources` for `sock` and for `reader`. It will make the code look more compact.